src/Diagnostics.RuntimeHost/Services/DevOpsClient/DevOpsClient.cs

```diff
@@ -173,27 +175,43 @@ public async Task<object> MakePullRequestAsync(string sourceBranch, string targe
 
             try
             {
-                dictionary[requestId + "--result"] = await gitClient.CreatePullRequestAsync(dictionary[requestId + "--pr"], _project, _repoID);
-            }
-            catch (Exception ex)
-            {
-                if (ex.Message.Contains("An active pull request for the source and target branch already exists"))
+                dictionary[requestId + "--prList"] = await GetPRListAsync(dictionary[requestId + "--source"], dictionary[requestId + "--target"], requestId);
+                if (dictionary[requestId + "--prList"].Count == 0)
                 {
-                    return new BadRequestObjectResult("An active pull request for the source and target branch already exists");
+                    dictionary[requestId + "--result"] = await gitClient.CreatePullRequestAsync(dictionary[requestId + "--pr"], _project, _repoID);
                 }
-                else {
-                    DiagnosticsETWProvider.Instance.LogDevOpsApiException(
-                    requestId,
-                    resourceUri,
-                    ex.Message,
-                    ex.GetType().ToString(),
-                    ex.StackTrace
-                    );
-                    throw;
+                else
+                {
+                    dictionary[requestId + "--result"] = dictionary[requestId + "--prList"][0];
```

Comment:
@rekhaswaminathan the change for handling duplicate PRs instead of throwing a BadRequest is here. If its already there i retrieve the existing pr and return that

Reply:
So this is where I initially asked about my question regarding request going to different instance.
Lets consider the following flow -
Author writes a new detector, Make Push + Create PR -> Lands in instance A & after PR creating, in-memory cache dictionary object now has the PR.
Now the author goes to make an edit to an existing branch which would need to update the active PR. This requests lands in instance B. Since you're checking against dictionary cache, how would this work?

Reply:
Also, I did not understand how appending requestid to the key works. The requestid keeps changing every time. Can you explain the purpose a bit more.

Reply:
all of the data retrieved from the dictionary is created in the same request it is being accessed by. the dictionary is only there so that multiple requests on the same instance wont change the data that other requests are still using. the values from the request on instance A wont be needed for the request on instance B